sdk/search/azure-search/src/samples/java/com/azure/search/SearchAsyncWithFullyTypedDocumentsExample.java

```diff
@@ -49,7 +49,7 @@ public static void main(String[] args) {
         SearchPagedFlux results = searchClient.search("searchText");
         results
             .subscribe(item -> {
-                Document document = item.getDocument();
+                Document document = new Document(item.getDocument());
```

Comment:
Customers shouldn't have to create `Document` instances explicitly.

Reply:
Depend on whether we want to have auto-gen code depends on manual codes.

Reply:
It should depend on what's best for the user experience.